hercules/emulator.py

```diff
@@ -145,9 +145,12 @@ def _read_external_data_file(self, filename):
         if "time" not in df_ext.columns:
             raise ValueError("External data file must have a 'time' column")
 
-        # Interpolate the external data according to time
+        # Interpolate the external data according to time.
+        # Goes to 1 time step past stoptime specified in the input file.
         times = np.arange(
-            self.helics_config_dict["starttime"], self.helics_config_dict["stoptime"], self.dt
+            self.helics_config_dict["starttime"],
+            self.helics_config_dict["stoptime"]+(2*self.dt),
```

Comment:
Changed this to create `times` variable that _includes_ the stoptime + one more timestep. `np.arange` is half-open, which does not include `end` and thus was causing errors when hercules was looking for a setpoint at the last timestep, but not finding it even though it was in the input `external_data_file`